browser/js/common/factories/user.factory.js

```diff
@@ -0,0 +1,43 @@
+app.factory('UserFactory', function($http) {
+
+	var fetchAll = function() {
+		return $http.get('/api/users')
+			.then(res => res.data);
+	};
+	var fetchOne = function(userId) {
+		return $http.get('/api/users/' + userId)
+			.then(res => res.data);
+	};
+
+	var deleteOne = function(user) {
+		return $http.delete('/api/users/' + user._id)
+			.then(res => res.data);
+	};
+
+	var promoteToAdmin = function(user) {
+		user.admin = true;
+		return $http.put('/api/users/' + user._id, user)
+			.then(res => res.data);
+	};
+
+	var demoteToUser = function(user) {
+		user.admin = false;
+		return $http.put('/api/users/' + user._id, user)
+			.then(res => res.data);
+	};
+
+	var resetPassword = function(user) {
+		user.requiresPasswordReset = true;
+		return $http.put('/api/users/' + user._id, user)
+			.then(res => res.data);
+	};
+
+	return {
+		fetchAll: fetchAll,
+		fetchOne: fetchOne,
+		deleteOne: deleteOne,
+		promoteToAdmin: promoteToAdmin,
+		demoteToUser: demoteToUser,
+		resetPassword: resetPassword
+	};
```

Comment:
An alternative to returning in this way is to instantiate an object at the beginning `var User = {}`, and then adding each function to the User object, and then returning the User object. It just might be more concise for moving forward if you ever have to add anything to the user factory